### PR TITLE
passky-desktop: init at 5.0.0

### PR DESCRIPTION
**Pull request by akkesm**
###### Motivation for this change

[Passky](https://github.com/Rabbit-Company/Passky-Desktop) is a cloud based password manager with an option for self-hosting, similar to Bitwarden. This is a package for the desktop client.

It's an electron app, I almost managed to build it from source but it ended up bugged. Wrapping the Appimage works.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -417,6 +417,16 @@
     github = "akho";
     githubId = 104951;
   };
+  akkesm = {
```
Comment (sikmir) [inline maintainers/maintainer-list.nix:420]: Adding yourself as a maintainer is preferred to be a separate commit.
Something like, `maintainers: add akkesm`. Then apply the passky-desktop commit.

**Comment by sikmir** [comment]
Add darwin support please:
```diff
diff --git i/pkgs/applications/misc/passky-desktop/default.nix w/pkgs/applications/misc/passky-desktop/default.nix
index edcd5ef5742..73abf202152 100644
--- i/pkgs/applications/misc/passky-desktop/default.nix
+++ w/pkgs/applications/misc/passky-desktop/default.nix
@@ -1,30 +1,22 @@
-{ lib, fetchurl, appimageTools }:
+{ lib, stdenv, fetchurl, appimageTools, undmg }:
 
 let
   pname = "passky-desktop";
   version = "5.0.0";
 
-  src = fetchurl {
-    url = "https://github.com/Rabbit-Company/Passky-Desktop/releases/download/v${version}/Passky-${version}.AppImage";
-    sha256 = "19sy9y2bcxrf10ifszinh4yn32q3032h3d1qxm046zffzl069807";
+  srcs = {
+    x86_64-linux = fetchurl {
+      url = "https://github.com/Rabbit-Company/Passky-Desktop/releases/download/v${version}/Passky-${version}.AppImage";
+      sha256 = "19sy9y2bcxrf10ifszinh4yn32q3032h3d1qxm046zffzl069807";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://github.com/Rabbit-Company/Passky-Desktop/releases/download/v${version}/Passky-${version}.dmg";
+      sha256 = "sha256-lclJOaYe+2XeKhJb2WcOAzjBMzK3YEmlS4rXuRUJYU0=";
+    };
   };
+  src = srcs.${stdenv.hostPlatform.system};
 
   appimageContents = appimageTools.extract { inherit pname version src; };
-in
-appimageTools.wrapType2 {
-  inherit pname version src;
-
-  extraInstallCommands = ''
-    mv $out/bin/${pname}-${version} $out/bin/${pname}
-
-    install -D ${appimageContents}/passky.desktop \
-      $out/share/applications/${pname}.desktop
-
-    substituteInPlace $out/share/applications/${pname}.desktop \
-      --replace 'Exec=AppRun' 'Exec=${pname}'
-
-    cp -r ${appimageContents}/usr/share/icons $out/share
-  '';
 
   meta = with lib; {
     homepage = "https://passky.org";
@@ -32,6 +24,38 @@ appimageTools.wrapType2 {
     downloadPage = "https://github.com/Rabbit-Company/Passky-Desktop/releases";
     license = licenses.gpl3Only;
     maintainers = with maintainers; [ akkesm ];
-    platforms = [ "x86_64-linux" ];
+    platforms = builtins.attrNames srcs;
   };
-}
+
+  linux = appimageTools.wrapType2 {
+    inherit pname version src meta;
+
+    extraInstallCommands = ''
+      mv $out/bin/${pname}-${version} $out/bin/${pname}
+
+      install -D ${appimageContents}/passky.desktop \
+        $out/share/applications/${pname}.desktop
+
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
+
+      cp -r ${appimageContents}/usr/share/icons $out/share
+    '';
+  };
+
+  darwin = stdenv.mkDerivation {
+    inherit pname version src meta;
+
+    nativeBuildInputs = [ undmg ];
+
+    sourceRoot = ".";
+
+    installPhase = ''
+      mkdir -p $out/Applications
+      cp -r *.app $out/Applications
+    '';
+  };
+in
+if stdenv.isDarwin
+then darwin
+else linux

```

**Comment by akkesm** [comment]
Done both things, unfortunately I can't test the darwin package.

Thanks for reviewing

**Review by sikmir — APPROVED** [review]
Builds and runs on darwin.

File: pkgs/applications/misc/passky-desktop/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    x86_64-linux = fetchurl {
+      url = "https://github.com/Rabbit-Company/Passky-Desktop/releases/download/v${version}/Passky-${version}.AppImage";
+      sha256 = "19sy9y2bcxrf10ifszinh4yn32q3032h3d1qxm046zffzl069807";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://github.com/Rabbit-Company/Passky-Desktop/releases/download/v${version}/Passky-${version}.dmg";
+      sha256 = "sha256-lclJOaYe+2XeKhJb2WcOAzjBMzK3YEmlS4rXuRUJYU0=";
+    };
+  };
+  src = srcs.${stdenv.hostPlatform.system};
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/passky-desktop/default.nix:17]: Missing or throw unsupported system.

Comment (akkesm) [reply pkgs/applications/misc/passky-desktop/default.nix]: Fixed.

**Comment by onny** [comment]
Result of `nixpkgs-review pr 148161` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>passky-desktop</li>
  </ul>
</details>

**Comment by onny** [comment]
Thank you!

**Comment by onny** [comment]
@akkesm Could you please update your keys in a separate PR?

**Comment by akkesm** [comment]
Done in #190012, sorry for the trouble.
